docs/fork_and_clone.md

```diff
@@ -0,0 +1,21 @@
+# Fork and clone the Autotune Repository
```

Comment:
Looking closely at the contents of this file, not sure if we need to have it as a separate file as it is just generic github usage HOWTO. I'd prefer if a shortened version of this is part of `CONTRIBUTING.md` itself